homeassistant/components/hue/bridge.py

```diff
@@ -26,8 +26,14 @@
 # How long should we sleep if the hub is busy
 HUB_BUSY_SLEEP = 0.5
 
-PLATFORMS_v1 = ["light", "binary_sensor", "sensor"]
-PLATFORMS_v2 = ["light", "binary_sensor", "sensor", "scene", "switch"]
+PLATFORMS_v1 = [Platform.LIGHT, Platform.BINARY_SENSOR, Platform.SENSOR]
+PLATFORMS_v2 = [
+    Platform.LIGHT,
+    Platform.BINARY_SENSOR,
+    Platform.SENSOR,
+    Platform.SCENE,
+    Platform.SWITCH,
+]
```

Comment:
Could these be in alphabetical order?
```suggestion
PLATFORMS_v1 = [Platform.BINARY_SENSOR, Platform.LIGHT, Platform.SENSOR]
PLATFORMS_v2 = [
    Platform.BINARY_SENSOR,
    Platform.LIGHT,
    Platform.SCENE,
    Platform.SENSOR,
    Platform.SWITCH,
]
```

Reply:
I haven't looked into the implementation, but I also wonder if these need to be lists (mutable) or if tuples (immutable) would be more appropriate.

Reply:
(Or `frozenset`s possibly even better semantically.)

Reply:
@scop Technically you're right, an immutable data type would work here, too (with one or two exceptions). However I don't think that would change much. From a (beginner) developer perspective it's much easier to write a list than say a frozenset. As for performance, they are created once and only read after that, so I wouldn't suspect they have much of an impact (if any at all).

My focus with these PRs has been to at least make sure all of them use the same, in this case `list`. That's also what was already preferred by most integrations. A limited number used tuples, but I've also seen something like this, which does work but isn't ideal IMO.
```py
PLATFORMS = "binary_sensor", "light", "sensor"
```

Reply:
Correctness is my concern here, nothing really to do with performance although it could have positive side effects there, too.

If the data is not supposed to be modified, which I don't know thoroughly and some exceptions were mentioned, tuples would be a (big enough) step towards the right direction. They are not harder to use than lists -- apart from needing the trailing comma for single element ones which I think is acceptable -- and they would eliminate one class of possible bugs. It is responsible to educate said (beginner) developers about correctness and set correct examples.

There's probably a lot of this across the codebase though, and even though if this whole PR set might make a few things worse by converting to list where tuple was used and appropriate (again, with the above mentioned caveat), with increased enum usage it is a net positive.

Reply:
I think that most contributors just look at what is used in the other components - especially beginners.
If we update the code everywhere to use the "correct" immutable type, then I don't think it would be a barrier to new contributors in any way.

Reply:
> Correctness is my concern here, nothing really to do with performance although it could have positive side effects there, too.
> 
> If the data is not supposed to be modified, which I don't know thoroughly and some exceptions were mentioned, tuples would be a (big enough) step towards the right direction.

IMO "It works" is sometimes good enough.


> They are not harder to use than lists -- apart from needing the trailing comma for single element ones which I think is acceptable -- and they would eliminate one class of possible bugs.

What class of bugs would they eliminate? Trailing commas are easy enough for us but when I think back to when I started with Python, missing them was always the worst kind of bug especially with a string. You just don't know what is wrong until you finally figure it out. IMO dealing with lists here is much easier.

> There's probably a lot of this across the codebase though, and even though if this whole PR set might make a few things worse by converting to list where tuple was used and appropriate (again, with the above mentioned caveat), with increased enum usage it is a net positive.

This isn't the first time the debate around lists and tuples has come up. It doesn't help either that in some instances Python is able to optimize it so that it doesn't matter which one you use. Both of these result in the same bytecode for example
```py
for x in (1, 2, 3):
    ...

for x in [1, 2, 3]:
    ...
```

If you want to change it, I won't stop you. I just think it's not really worth it in the end.

Reply:
> What class of bugs would they eliminate?

Accidental modification of existing collection elements, and adding to/removing from collections that should not be modified. Varies case by case how likely it is to happen, how bad it is when it does happen, and how likely it is to go unnoticed in reviews or otherwise. The case at hand is probably not in the worst end of any of those categories.

Re missing commas, tuples and strings: mypy can help. I'm hopeful we can extend its usage sometime to cover also the hypothetical case that is being discussed here. 

> It doesn't help either that in some instances Python is able to optimize it [...]

Maybe. But as said, correctness is what I'm personally interested in here, not performance. Ditto in general first. (FWIW from correctness point of view there's no difference in the given example cases as far as I'm concerned, and for them I'd let other more interested parties discuss the possible performance differences.)